src/annotate/celltypist/test.py

```diff
@@ -159,49 +194,116 @@ def test_with_model(run_component, random_h5mu_path):
 
-    # fails because input data are not lognormalized
+    # fails because input data are not correctly lognormalized
     with pytest.raises(subprocess.CalledProcessError) as err:
         run_component(
             [
                 "--input",
                 input_file,
+                "--input_layer",
+                "log_normalized",
                 "--reference",
                 reference_file,
+                "--reference_layer",
+                "log_normalized",
                 "--reference_var_gene_names",
                 "ensemblid",
                 "--output",
                 output_file,
             ]
         )
     assert re.search(
-        r"Invalid expression matrix, expect log1p normalized expression to 10000 counts per cell",
+        r"Invalid expression matrix, expect input layer to contain log1p normalized expression to 10000 counts per cell",
         err.value.stdout.decode("utf-8"),
     )
 
-    # fails because reference data are not lognormalized
-    with pytest.raises(subprocess.CalledProcessError) as err:
-        run_component(
-            [
-                "--input",
-                input_file,
-                "--layer",
-                "log_normalized",
-                "--reference",
-                reference_file,
-                "--reference_var_gene_names",
-                "ensemblid",
-                "--output",
-                output_file,
-            ]
-        )
-    assert re.search(
-        r"Invalid expression matrix, expect log1p normalized expression to 10000 counts per cell",
-        err.value.stdout.decode("utf-8"),
+
+def test_with_hvg(
+    run_component, random_h5mu_path, write_mudata_to_file, input_mdata, reference_mdata
+):
+    output_file = random_h5mu_path()
+    input_file = write_mudata_to_file(input_mdata)
+    reference_file = write_mudata_to_file(reference_mdata)
+
+    run_component(
+        [
+            "--input",
+            input_file,
+            "--reference",
+            reference_file,
+            "--reference_obs_target",
+            "cell_ontology_class",
+            "--reference_var_gene_names",
+            "ensemblid",
+            "--reference_var_input",
+            "highly_variable",
+            "--output",
+            output_file,
+        ]
+    )
+
+    assert os.path.exists(output_file), "Output file does not exist"
+
+    input_mudata = mu.read_h5mu(input_file)
+    output_mudata = mu.read_h5mu(output_file)
+
+    assert_annotation_objects_equal(input_mudata.mod["prot"], output_mudata.mod["prot"])
+
+    assert {"celltypist_pred", "celltypist_probability"}.issubset(
+        output_mudata.mod["rna"].obs.keys()
+    ), "Required keys not found in .obs"
+
+    obs_values = output_mudata.mod["rna"].obs["celltypist_probability"]
+    assert all(0 <= value <= 1 for value in obs_values), (
+        ".obs at celltypist_probability has values outside the range [0, 1]"
+    )
+
+
+def test_duplicate_categorical_index_entry(
```

Comment:
This test does not seem to be failing when I remove the `.astype(str)` in `src/utils/set_var_index.py`. But I think it should?